### PR TITLE
[Cortex REST API] New Script GetIncidentTasks

### PR DESCRIPTION
**Pull request by rundssoar**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] Ready

## Description
This is a python port of the "GetIncidentTasksByState" Automation with the addition, that it is now possible to get Incident tasks by tag, names, states or any combination of those

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+    entry = CommandResults(outputs_prefix='Tasks',
+                           outputs=res,
+                           readable_output=tableToMarkdown(
+                               f'{format_title(name, tag,args.get("states", ""))}(Incident  # {inc_id})',
+                               res,
+                               ['id', 'name', 'state', 'owner', 'scriptId']),
+                           entry_type=EntryType.NOTE)
+
+    return entry
```
Comment (shmuel44) [inline Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py:129]:
```suggestion
    return CommandResults(
        outputs_prefix='Tasks',
        outputs_key_field='id',
        outputs=res,
        readable_output=tableToMarkdown(
            f'{format_title(name, tag, args.get("states", ""))}(Incident  # {incident_id})',
            res,
            ['id', 'name', 'state', 'owner', 'scriptId']
        )
    )
```

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    return f'Tasks with {"and ".join(items)}'.strip()
+
+
+''' COMMAND FUNCTION '''
+
+
+def get_task_by_state_command(args: dict[str, Any]) -> CommandResults:
+    name = args.get('name', None)
+    tag = args.get('tag', None)
+    states = get_states(args.get('states', ''))
```
Comment (shmuel44) [inline Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py:89]: Please use `argToList` function
```suggestion
    states = get_states(argToList(args.get('states')))
```

**Review by shmuel44 — COMMENTED** [review]
Hi @rundssoar,
Thank you for your contribution!
Good work :)

please add docstring using [Google Style](https://google.github.io/styleguide/pyguide.html#:~:text=Yes%3A%0A%20%20def%20connect_to_next_port(self%2C%20minimum%3A%20int)%20%2D%3E%20int%3A%0A%20%20%20%20%22%22%22Connects%20to%20the%20next%20available%20port.%0A%0A%20%20%20%20Args%3A%0A%20%20%20%20%20%20minimum%3A%20A%20port%20value%20greater%20or%20equal%20to%201024.%0A%0A%20%20%20%20Returns%3A%0A%20%20%20%20%20%20The%20new%20minimum%20port.%0A%0A%20%20%20%20Raises%3A%0A%20%20%20%20%20%20ConnectionError%3A%20If%20no%20available%20port%20is%20found.%0A%20%20%20%20%22%22%22%0A%20%20%20%20if%20minimum) for each function,
and add more unit test
You can use the following guide - https://xsoar.pan.dev/docs/integrations/unit-testing as well as examples of unit tests in the HelloWorld integration and other packs from the content repository.

Please feel free to reach out to me with any questions
I'm available here or on Slack :)
Thanks again 🙏

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+}
+
+''' STANDALONE FUNCTION '''
+
+
+def map_to_array(m: dict) -> list:
+    arr = []
+    for k in m:
+        arr.append(m[k])
+    return arr
```
[the remark this answers is not in the document]
Comment (rundssoar) [reply Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py:25]: This was left over from doing a simple port of the Javascript code

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+
+def get_sub_playbook_tasks(tasks: list) -> list:
+    readyTasks = []
```
[the remark this answers is not in the document]
Comment (rundssoar) [reply Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py:29]: This was lefft over from the Javascript port

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+
+''' COMMAND FUNCTION '''
+
+
+def get_task_by_state_command(args: dict[str, Any]) -> CommandResults:
+    name = args.get('name', None)
+    tag = args.get('tag', None)
+    states = get_states(args.get('states', ''))
+    inc_id = args.get('inc_id')
```
[the remark this answers is not in the document]
Comment (rundssoar) [reply Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py:90]: the parameter is called inc_id in the yml, since that is what it was called in the JavaScript version.

I prefer to do proper error handling, but it seems common practice to access required params directly

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    for k in m:
+        arr.append(m[k])
+    return arr
+
+
+def get_sub_playbook_tasks(tasks: list) -> list:
+    readyTasks = []
+    for task in tasks:
+        if task.get('type') == 'playbook' and task.get('subPlaybook'):
+            readyTasks.extend(get_sub_playbook_tasks(map_to_array(task.get('subPlaybook', {}).get('tasks'))))
```
[the remark this answers is not in the document]
Comment (rundssoar) [reply Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py:32]: sounds reasonable

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+        raise Exception(res)
+        # return_error(res)
+
+    workplan = res[0].get('Contents', {}).get('response', {}).get('invPlaybook')
+    if not workplan or not workplan.get('tasks', None) or len(workplan.get('tasks', [])) == 0:
+        return CommandResults(outputs_prefix='Tasks', outputs=[],
+                              readable_output=f'Workplan for incident {inc_id} has no tasks.')
+
+    tasks = map_to_array(workplan.get('tasks'))
+    allTasks = get_all_playbook_tasks(tasks)
```
[the remark this answers is not in the document]
Comment (rundssoar) [reply Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py:102]: That one was tricky, since i coulnd't find any documentation on the dq JavaScript function, so i was guessing, what it was even doing.

[the remark this answers is not in the document]
Comment (shmuel44) [reply Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py:102]: Nice!

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (shmuel44) [reply Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py:90]: When the argument is marked as `required: true` in the `yml` file, we don't need to use `get()` .

**Comment by rundssoar** [comment]
@shmuel44 I should have addressed all your comments now. Let me know, if there is something left.

**Comment by shmuel44** [comment]
@rundssoar It looks good!
We have decided to deprecate the `GetIncidentTasksByState` script and use your script instead.
Could you please run the following command in your terminal?

`demisto-sdk format --deprecate -i Packs/DemistoRESTAPI/Scripts/GetIncidentTasksByState/GetIncidentTasksByState.yml`

This will format the code and deprecate the `GetIncidentTasksByState` script. 
After running the above command, please change the `comment` field in the `GetIncidentTasksByState.yml` to `Deprecated. Use the GetIncidentTasks script instead.`

Once you have made these changes, please run the following command in your terminal:

`demisto-sdk update-release-notes`

Also, could you please add a docstring for the unit test function using the [given, when, then ](https://github.com/demisto/content/blob/master/Packs/AWS_SystemManager/Integrations/AWSSystemManager/AWSSystemManager_test.py#L546.)format? 

Let me know if you have any questions or need further assistance.

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks_test.py [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+@pytest.mark.parametrize('args', [({'name': 'task name'})])
+def test_get_task_command_missing(args):
+    with pytest.raises(KeyError):
+        get_task_command(args)
+
+
+@pytest.mark.parametrize('args', [({'inc_id': '1', 'name': 'task name'})])
+def test_get_task_command_failure(args):
+    with pytest.raises(Exception):
+        get_task_command(args)
```
Comment (shmuel44) [inline Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks_test.py:100]: Please note that there is no need to test the `args`. The `args` have already been tested, and the server should always return a dictionary. 

Could you please create a unit test for the `get_task_command` function and use the `mocker` fixture to mock the response from the `demisto.executeCommand('core-api-get', {'uri': f'/investigation/{inc_id}/workplan'})` command? 

Here's an [example](https://github.com/demisto/content/blob/master/Packs/AWS-Enrichment-Remediation/Scripts/AWSRecreateSG/AWSRecreateSG_test.py#L58) of how to use `mocker` to patch the `executeCommand` function:

```python
mocker.patch.object(demisto, "executeCommand", return_value=INCIDENT_TASK)
```

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+                'completedDate': task.get('completedDate'),
+                'parentPlaybookID': task.get('parentPlaybookID'),
+                'completedBy': task.get('completedBy')
+            })
+
+    return CommandResults(outputs_prefix='Tasks',
+                          outputs_key_field='id',
+                          outputs=res,
+                          readable_output=tableToMarkdown(
+                              f'{format_title(name, tag,args.get("states", ""))}(Incident  # {inc_id})',
```
Comment (shmuel44) [inline Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py:148]:
```suggestion
                               f'Incident #{inc_id} Playbook Tasks',
```
Let's use this format to comply with current conventions.

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.yml [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+  description: Incident ID to get tasks from.
+- name: states
+  isArray: true
+  description: 'Comma separated list of states. Possible values: New, InProgress, Completed, Waiting, Error, Skipped, Blocked (leave empty to get all tasks).'
+- name: name
+  description: 'The name of the Task to search.'
+  type: string
+- name: tag
+  description: 'The tag to search.'
+  type: string
```
Comment (shmuel44) [inline Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.yml:25]:
```suggestion
- name: tag
  description: 'The tag to search.'
```
It's not necessary to specify the `type` when the `type` is `string`

**Comment by rundssoar** [comment]
I've added the mocker test, and all the test should have the docstrings
I also ran the commands to deprecate the old script as requested

File: Tests/conf.json [hunk 11]
```diff
@@ -6264,4 +6264,4 @@
         "IPv6 CIDR extraction test",
         "IPv6 extraction test"
     ]
-}
+}
```
Comment (shmuel44) [inline Tests/conf.json:6267]: Please revert the changes in this file.
```suggestion
}

```

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.yml [hunk 12]
```diff
@@ -0,0 +1,65 @@
+commonfields:
+  id: GetIncidentTasks
+  version: -1
+name: GetIncidentTasks
+script: ''
+type: python
+subtype: python3
+tags:
+- Utility
+comment: Get all tasks for specific incident by the given state, name and/or tag.
```
Comment (ShirleyDenkberg) [inline Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.yml:10]:
```suggestion
comment: Get all tasks for a specific incident by the given state, name and/or tag.
```

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.yml [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+enabled: true
+args:
+- name: inc_id
+  required: true
+  default: true
+  description: Incident ID to get tasks from.
+- name: states
+  isArray: true
+  description: 'Comma separated list of states. Possible values: New, InProgress, Completed, Waiting, Error, Skipped, Blocked (leave empty to get all tasks).'
```
Comment (ShirleyDenkberg) [inline Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.yml:19]:
```suggestion
  description: 'Comma-separated list of states. Possible values: New, InProgress, Completed, Waiting, Error, Skipped, Blocked. (Leave empty to get all tasks).'
```

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.yml [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+- name: name
+  description: 'The name of the Task to search.'
```
Comment (ShirleyDenkberg) [inline Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.yml:21]:
```suggestion
  description: 'The name of the task to search.'
```

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.yml [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+  description: Task start date.
+  type: unknown
+- contextPath: Tasks.completedDate
+  description: Task completed date.
+  type: unknown
+- contextPath: Tasks.dueDate
+  description: Task due date (SLA).
+  type: unknown
+- contextPath: Tasks.parentPlaybookID
+  description: Task parent playbook ID (in case the task is part of sub-playbook).
```
Comment (ShirleyDenkberg) [inline Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.yml:55]:
```suggestion
  description: Task parent playbook ID (in case the task is part of a sub-playbook).
```

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/README.md [hunk 16]
```diff
@@ -0,0 +1,40 @@
+Gets all tasks for a specific incident by the given state, name or tag.
+If multiple values are given, only tasks matching all of them are returned. 
+For example, if name and states is given, only tasks with the given name in a given state are returned
```
Comment (ShirleyDenkberg) [inline Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/README.md:3]:
```suggestion
For example, if name and state is given, only tasks with the given name in a given state are returned.
```

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/README.md [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
+| Script Type | python |
+| Tags | Utility |
+
+
+## Inputs
+---
+
+| **Argument Name** | **Description** |
+| --- | --- |
+| inc_id | The Incident ID to get tasks from. |
```
Comment (ShirleyDenkberg) [inline Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/README.md:19]:
```suggestion
| inc_id | The incident ID to get tasks from. |
```

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/README.md [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+| states | The comma-separated list of states. Can be, "New", "InProgress", "Completed", "Waiting", "Error", "Skipped", "Blocked". You can also leave this field empty to get all tasks. |
+| name | The name of the Task to search. |
+| tag | The tag of the Task to search
```
Comment (ShirleyDenkberg) [inline Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/README.md:22]:
```suggestion
| name | The name of the task to search. |
| tag | The tag of the task to search.
```

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/README.md [hunk 19]
```diff
@@ -0,0 +1,40 @@
...
+| Tasks.id | The task ID. | string |
+| Tasks.name | The task name. | string |
+| Tasks.type | The type of the task. Can be, "regular", "condition", "title", "playbook", or "start". | string |
+| Tasks.owner | The task owner. | string |
+| Tasks.state | The task state. Can be, "inprogress", "Completed", "WillNotBeExecuted", "Error", "Waiting", "Blocked", and empty string for not started. | string |
+| Tasks.scriptId | The task related script (empty if manual). | string |
+| Tasks.startDate | The task start date. | unknown |
+| Tasks.completedDate | The task completed date. | unknown |
+| Tasks.dueDate | The task due date (SLA). | unknown |
+| Tasks.parentPlaybookID | The task parent playbook ID (in case the task is part of sub-playbook). | unknown |
```
Comment (ShirleyDenkberg) [inline Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/README.md:39]:
```suggestion
| Tasks.parentPlaybookID | The task parent playbook ID (in case the task is part of a sub-playbook). | unknown |
```

**Comment by ShirleyDenkberg** [comment]
@shmuel44 @sapirshuker Doc review completed.

**Comment by rundssoar** [comment]
@shmuel44 Let me know, if there is anything left to change.
The conflict should be easily resolvable, as i've updated the release notes according

File: Packs/DemistoRESTAPI/ReleaseNotes/1_3_47.md [hunk 20]
```diff
@@ -5,3 +5,4 @@
 
 - Added support for running on multiple entry IDs by using the *isArray: true* configuration.
 - Updated the Docker image to: *demisto/python3:3.10.13.86272*.
+
```
Comment (shmuel44) [inline Packs/DemistoRESTAPI/ReleaseNotes/1_3_47.md:8]:
```suggestion
```

**Comment by shmuel44** [comment]
Hi @rundssoar,
the code looks good!

We're ready for a demo. Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me know when you're available for one over DFIR.

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py [hunk 21]
```diff
@@ -0,0 +1,146 @@
...
+                'completedBy': task.get('completedBy')
+            })
+
+    return CommandResults(outputs_prefix='Tasks',
+                          outputs_key_field='id',
+                          outputs=res,
+                          readable_output=tableToMarkdown(
+                              f'Incident #{inc_id} Playbook Tasks',
+                              res,
+                              ['id', 'name', 'state', 'owner', 'scriptId']))
```
Comment (shmuel44) [inline Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py:129]:
```suggestion
                          readable_output=tableToMarkdown(
                              name=f'Incident #{inc_id} Playbook Tasks',
                              t=res,
                              headers=['id', 'name', 'state', 'owner', 'scriptId'],
                              removeNull=True,
                              )
                          )
```

Comment (rundssoar) [reply Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py]: done

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.yml [hunk 22]
```diff
@@ -0,0 +1,65 @@
...
+comment: Get all tasks for a specific incident by the given state, name and/or tag.
+enabled: true
+args:
+- name: inc_id
+  required: true
+  default: true
+  description: Incident ID to get tasks from.
+- name: states
+  isArray: true
+  description: 'Comma-separated list of states. Possible values: New, InProgress, Completed, Waiting, Error, Skipped, Blocked. (Leave empty to get all tasks).'
```
Comment (shmuel44) [inline Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.yml:19]:
```suggestion
  description: 'Comma-separated list of states. Possible values: New, InProgress, Completed, Waiting, Error, LoopError, Skipped, Blocked. (Leave empty to get all tasks).'
```

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.yml [hunk 23]
```diff
@@ -0,0 +1,65 @@
...
+  description: Task name.
+  type: string
+- contextPath: Tasks.type
+  description: The type of the task (regular, condition, title, playbook, start).
+  type: string
+- contextPath: Tasks.owner
+  description: Task owner.
+  type: string
+- contextPath: Tasks.state
+  description: Task state (inprogress, Completed, WillNotBeExecuted, Error, Waiting, Blocked, and empty string for not started).
```
Comment (shmuel44) [inline Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.yml:40]:
```suggestion
  description: Task state (inprogress, Completed, WillNotBeExecuted, Error, LoopError, Waiting, Blocked, and empty string for not started).
```

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/README.md [hunk 24]
```diff
@@ -0,0 +1,40 @@
...
+---
+
+| **Path** | **Description** | **Type** |
+| --- | --- | --- |
+| Tasks | The entire task object. | Unknown |
+| Tasks.id | The task ID. | string |
+| Tasks.name | The task name. | string |
+| Tasks.type | The type of the task. Can be, "regular", "condition", "title", "playbook", or "start". | string |
+| Tasks.owner | The task owner. | string |
+| Tasks.state | The task state. Can be, "inprogress", "Completed", "WillNotBeExecuted", "Error", "Waiting", "Blocked", and empty string for not started. | string |
```
Comment (shmuel44) [inline Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/README.md:34]:
```suggestion
| Tasks.state | The task state. Can be, "inprogress", "Completed", "WillNotBeExecuted", "Error", "LoopError", "Waiting", "Blocked", and empty string for not started. | string |
```

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/README.md [hunk 25]
```diff
@@ -0,0 +1,40 @@
...
+| Tags | Utility |
+
+
+## Inputs
+---
+
+| **Argument Name** | **Description** |
+| --- | --- |
+| inc_id | The incident ID to get tasks from. |
+| states | The comma-separated list of states. Can be, "New", "InProgress", "Completed", "Waiting", "Error", "Skipped", "Blocked". You can also leave this field empty to get all tasks. |
```
Comment (shmuel44) [inline Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/README.md:20]:
```suggestion
| states | The comma-separated list of states. Can be, "New", "InProgress", "Completed", "Waiting", "Error", "LoopError", "Skipped", "Blocked". You can also leave this field empty to get all tasks. |
```

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py [hunk 26]
```diff
@@ -0,0 +1,147 @@
...
+
+
+TASK_STATES = {
+    'new': '',
+    'inprogress': 'inprogress',
+    'completed': 'Completed',
+    'waiting': 'Waiting',
+    'error': 'Error',
+    'looperror': 'LoopError',
+    'skipped': 'WillNotBeExecuted',
```
Comment (shmuel44) [inline Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py:14]: After I spoke with @AdiPeret, we decided that we should add the 'WillNotBeExecuted' as a possible value.

```suggestion
    'skipped': 'WillNotBeExecuted',
    'WillNotBeExecuted': 'WillNotBeExecuted',
```

Comment (rundssoar) [reply Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py:14]: OK, i'm adding it

File: Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py [hunk 27]
```diff
@@ -0,0 +1,147 @@
...
+def get_states(states: list[str]) -> list[str]:
+    """Get the internal states names for the given search states
+
+    Args:
+        states (list[str]): a list of states to search
+
+    Returns:
+        list[str]: a list of the internal names of the states given in the input or all possible states, if input was empty
+    """
+    if 'error' in states:
```
Comment (rundssoar) [inline Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py:53]: @shmuel44 Shall we leave this in here, or remove it, since we would have the LoopError as a dedicated possible stat

Comment (shmuel44) [reply Packs/DemistoRESTAPI/Scripts/GetIncidentTasks/GetIncidentTasks.py:54]: We have decided to keep it the same as it was in the old script.

File: Packs/DemistoRESTAPI/ReleaseNotes/1_3_48.md [hunk 28]
```diff
@@ -1,11 +1,10 @@
 
 #### Scripts
 
+
```
Comment (shmuel44) [inline Packs/DemistoRESTAPI/ReleaseNotes/1_3_48.md:4]:
```suggestion
```

File: Packs/DemistoRESTAPI/ReleaseNotes/1_3_48.md [hunk 29]
```diff
@@ -7,5 +7,3 @@
 
 ##### GetTasksWithSections
 - Updated the Docker image to: *demisto/python3:3.10.13.86272*.
```
Comment (shmuel44) [inline Packs/DemistoRESTAPI/ReleaseNotes/1_3_48.md:9]:
```suggestion
- Updated the Docker image to: *demisto/python3:3.10.13.86272*.


```

**Comment by shmuel44** [comment]
Hey @rundssoar, could you please fix the unit test?

**Review by shmuel44 — APPROVED** [review]
@rundssoar Good Work!

**Comment by shmuel44** [comment]
Hi @rundssoar ,
I merged the internal PR, and your update is now in the master branch of the content repo.
Your update should be available in the marketplace within the next 24 hours.
Again, thank you very much for your contribution!
